clients/automation/automation.go

```diff
@@ -218,20 +224,22 @@ func (a Client) List(ctx context.Context, resourceType automation.ResourceType)
 			continue
 		}
 
+		body, err := io.ReadAll(resp.Body)
+		resp.Body.Close()
+		if err != nil {
+			logr.FromContextOrDiscard(ctx).Error(err, bodyReadErrMsg)
+			return ListResponse{}, api.NewAPIErrorFromResponseAndBody(resp, body)
+		}
+
 		// if one of the response has code != 2xx return empty list with response info
 		if !rest.IsSuccess(resp) {
-			body, err := io.ReadAll(resp.Body)
-			resp.Body.Close()
-			if err != nil {
-				return ListResponse{}, err
-			}
-
 			return ListResponse{}, api.NewAPIErrorFromResponseAndBody(resp, body)
 		}
 
 		result, err = unmarshalJSONList(resp)
 		if err != nil {
-			return ListResponse{}, fmt.Errorf("failed to parse list response:%w", err)
+			logr.FromContextOrDiscard(ctx).Error(err, "failed to unmarshal json response")
```

Comment:
I wonder if we should be returning this as an API error - that feels a bit weird for a 2xx status code.

Reply:
due to lack of significantly better alternatives, i think it is ok
returning `responseVal, err` requires different handling on the client side,
leaving it like it is, hides the http response code,
introducing another type of error just for that seems awkward....